### PR TITLE
Facet Search API

### PR DESCRIPTION
**Pull request by gmourier**
🤖  [API Diff](https://github.com/meilisearch/specifications/pull/246#issuecomment-1603911181)

---

# Summary

Describes the facet search endpoint allowing to search for facet values

---

# Changes

- Introduces facet-search-api.md
- Introduces new errors definition
     - `missing_facet_search_facet_name`
     - `invalid_facet_search_facet_name`
     - `invalid_facet_search_facet_query`
- Describes the new API endpoint in open-api.yaml
- Describes added telemetry

## Misc

- Fix `wordsMatchingStrategy` to `matchingStrategy` in open-api.yml

# Out Of Scope

N/A

---

# Attention To Reviewers

N/A
---

## Misc

- [x] Update OpenAPI specification file
- [x] Update telemetry datapoints

### CODE REVIEW

File: text/0034-telemetry-policies.md [hunk 1]
```diff
@@ -112,10 +113,10 @@ The collected data is sent to [Segment](https://segment.com/). Segment is a plat
 | `stats.indexes_number`                  | Number of indexes                                       | 2                 | Every hour |
 | `start_since_days`                      | Number of days since instance was launched              | 365               | Every hour |
 | `user_agent`                            | User-agent header encountered during one or more API calls | ["Meilisearch Ruby (v2.1)", "Ruby (3.0)"] | `Documents Searched POST`, `Documents Searched GET`, `Index Created`, `Index Updated`, `Documents Added`, `Documents Updated`, `Documents Deleted`, `Settings Updated`, `Ranking Rules Updated`, `SortableAttributes Updated`, `FilterableAttributes Updated`, `SearchableAttributes Updated`, `TypoTolerance Updated`, `Pagination Updated`, `Faceting Updated`, `DistinctAttribute Updated`, `DisplayedAttributes Updated`, `StopWords Updated`, `Synonyms Updated`, `Dump Created`, `Tasks Seen`, `Stats Seen`, `Health Seen`, `Version Seen`, `Documents Searched by Multi-Search POST` |
```
Comment (gmourier) [inline text/0034-telemetry-policies.md:115]:
```suggestion
| `user_agent`                            | User-agent header encountered during one or more API calls | ["Meilisearch Ruby (v2.1)", "Ruby (3.0)"] | `Documents Searched POST`, `Documents Searched GET`, `Index Created`, `Index Updated`, `Documents Added`, `Documents Updated`, `Documents Deleted`, `Settings Updated`, `Ranking Rules Updated`, `SortableAttributes Updated`, `FilterableAttributes Updated`, `SearchableAttributes Updated`, `TypoTolerance Updated`, `Pagination Updated`, `Faceting Updated`, `DistinctAttribute Updated`, `DisplayedAttributes Updated`, `StopWords Updated`, `Synonyms Updated`, `Dump Created`, `Tasks Seen`, `Stats Seen`, `Health Seen`, `Version Seen`, `Documents Searched by Multi-Search POST`, `Facet Searched POST` |
```

**Review by gmourier — COMMENTED** [review]
Adding missing `Facet Searched POST` event on the `user-agent` telemetry node

File: text/0246-facet-search-api.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
+# Facet Search API
+
+## 1. Summary
+
+The facet search endpoint retrieves facet values from a query.
+
+## 2. Motivation
+
+When many values exist for a facet, end-users need to be able to discover non-show values they can select in order to refine their faceted search. Being able to search for values for a facet solves this need.
```
Comment (bidoubiwa) [inline text/0246-facet-search-api.md:9]:
```suggestion
When a facet has lots of values, end-users need to be able to find specific values that are not returned if there are too many or lost in a sea of other values. Being able to search for specific values of a facet solves this need.
```

File: text/0246-facet-search-api.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+- 🔴 Sending a value with a different type than `String` for `facetName` returns a [missing_facet_search_facet_name](0061-error-format-and-definitions.md#minvalid_facet_search_facet_name) error.
+- 🔴 Sending a field not defined as a `filterableAttributes` for `facetname` returns an [invalid_facet_search_facet_name](0061-error-format-and-definitions.md#invalid_facet_search_facet_name) error.
+
+#### 3.1.2. `facetQuery`
+
+- Type: String
+- Required: False
+- Default: `null`
+
+`facetQuery` contains the terms to search within the facet values contained in the index documents.
```
Comment (bidoubiwa) [inline text/0246-facet-search-api.md:58]:
```suggestion
`facetQuery` contains the terms to search within the facet values of the `facetName`.
```

**Review by maryamsulemani97 — APPROVED** [review]

**Review by bidoubiwa — APPROVED** [review]

**Review by gillian-meilisearch — APPROVED** [review]

**Pull request merged**